ipld/read.go

```diff
@@ -88,7 +87,31 @@ func GetLeaf(ctx context.Context, dag ipld.NodeGetter, root cid.Cid, leaf, total
 	return GetLeaf(ctx, dag, root, leaf, total)
 }
 
-// GetLeavesByNamespace returns all the shares from the given DataAvailabilityHeader root
+// GetSharesByNamespace returns all the shares from the given root
+// with the given namespace.ID.
+func GetSharesByNamespace(
+	ctx context.Context,
+	dag ipld.NodeGetter,
+	root cid.Cid,
+	nID namespace.ID,
+) ([]Share, error) {
+	leaves, err := GetLeavesByNamespace(ctx, dag, root, nID)
+	if err != nil {
+		return nil, err
+	}
+
+	shares := make([]Share, len(leaves))
+	for i, leaf := range leaves {
+		shares[i] = leafToShare(leaf)
+	}
+
+	return shares, nil
+}
+
+// ErrNotFoundInRange is returned when namespace.ID is not found under the root.
+var ErrNotFoundInRange = fmt.Errorf("ipld: namespace.ID not found in range")
```

Comment:
I'm a bit confused here TBH. 
Personally, I would prefer to just return `nil` slice instead of error. Not finding namespace is normal (not exceptional/erroneous) situation IMHO.

Internally this error is used only here: https://github.com/celestiaorg/celestia-node/pull/661/files#diff-213b77ca67ac98ceba2a0a80fafb1cd3b312906cb991a5003c808e673160f5f0R140-R147

But if we return `nil, nil` this will streamline the code to something like: 
```go
nds, err := GetLeavesByNamespace(ctx, dag, lnk.Cid, nID)
if err != nil {
	return nil, err
}
out = append(out, nds...)
```
Extra notes:
 * currently error handling is broken anyways (in case of actual error (from dag, if `err != nil` and `err != ErrNotFoundInRange`))
 * `errors.Is` could be used

Reply:
@renaynay, well, you originally introduced this error, so I am not sure of the original intention here. 

According to the code, the answer is yes.
>corresponds to it not being found inside a specific row right

Reply:
@tzdybal, I am fine with removing this error and making it a nil case. Although, you can think of this case from the `io.EOF` perspective, something like `EOData`, where there is no more data to read or no data under the root.

Reply:
```suggestion
var ErrNotFoundInRoot = fmt.Errorf("ipld: namespace.ID not found in root")
```

can we do this then? The word `range` is confusing now. (I know i intro'd this err)

Reply:
> Personally, I would prefer to just return nil slice instead of error. Not finding namespace is normal (not exceptional/erroneous) situation IMHO.
If this is is the only case, where the method would return `nil` and no error, I think it makes sense to remove the error completely. IIRC the original intention was to signal that the namespace holds no data in an unambiguous and handleable way.

Reply:
> Personally, I would prefer to just return nil slice instead of error. Not finding namespace is normal (not exceptional/erroneous) situation IMHO.
If this is is the only case, where the method would return `nil` and no error, I think it makes sense to remove the error completely. IIRC the original intention was to signal that the namespace holds no data in an unambiguous and handleable way.

Reply:
Agree with removal. Lets do it

Reply:
Removed. Looks good without it. Thanks for the suggestion @tzdybal and pls resolve the comment